Human update_icon refactor

comma opened the pull request:
Moves overlay generation into items. This reduces some copypaste code.

comma committed:
Clean up of human overlays generation

Items now generate their overlays instead of the pasta restraunt in update_icon.dm.

comma committed:
List of default onmob icons

Instead of defining default icon per type.
This way items that can be worn on different slots are handled naturally.
Converts suit storage overlay icon to new system (previously didn't want to because uggh another special icon file setting)

comma committed:
Removes some defines that overlapped with new list

comma committed:
Removes duplicate rollsleeves verb

While I'm in the area, why not

comma commented:
whoops looks like I forgot to add proper bloodmask states to all the clothing items

comma commented:
I think I can squeeze it some more

comma committed:
Moved hud updates out of icon update

HUDs to do not hold hands cue special setting for them in inventory.dm
Also moved some inventory related procs for humans to proper file. Made putting things in hands use equip_to_slot instead of duplicating code.

comma committed:
Forgot glasses are snowflakes too

Hate item_state/icon_state duality, but too deep rooted to fix it without maybe a mass rename of icon_states.

comma committed:
Moved various unrelated code out

Leg and hand cuffs are now handled in equipping proc.
Straightjackets handle their special effect.

comma committed:
Bits and bobs

Collars are now generated suit-side too. Since they are layered differently, moved them to separate image.
Fire nad sugery overlays are not painted by mob color too.
Added default icon_state for ID slot, since there /is/ pretty much only one in that dmi. And getting wrong one means fullsized naked man as your ID overlay.

comma commented:
Guess that's it

Fox-McCloud commented:
Does this or will you attempt to do anything to help with the insane cost associated with `update_icons`?

comma commented:
No.

Fox-McCloud commented:
Alrighty. Thank ya.

RemieRichards commented:
Throw out the `/icon`s dang it.
that's your issue.

comma commented:
One day, one day. Too much effort, gotta redo every mob sprite bit etc.

PsiOmegaDelta commented on code/modules/clothing/clothing.dm:
```diff
@@ -23,6 +23,22 @@
+		if(blood_DNA)
+			var/image/bloodsies	= image(icon = user_human.species.blood_mask, icon_state = blood_overlay_type)
+			bloodsies.color		= blood_color
+			bloodsies.appearance_flags = RESET_COLOR
```
Shouldn't this use `overlay_image()`?

comma replied:
ya, missed it

PsiOmegaDelta commented on code/modules/mob/living/carbon/human/update_icons.dm:
```diff
@@ -1069,22 +732,21 @@ var/global/list/damage_icon_parts = list()
 	if(on_fire)
-		overlays_standing[FIRE_LAYER] = image("icon"='icons/mob/OnFire.dmi', "icon_state"="Standing", "layer"=FIRE_LAYER)
+		var/image/standing = image("icon"='icons/mob/OnFire.dmi', "icon_state"="Standing", "layer"=FIRE_LAYER)
+		standing.appearance_flags = RESET_COLOR
```
Shouldn't this use `overlay_image()`?

comma replied:
Need to set layer, so either way one var edit outside of constructor would be used.

PsiOmegaDelta replied:
But that means overlay_image() doesn't have the same arguments as image() :<
Even then, could still use overlay_image() and set the layer for the returned image instead of the appearance_flag.

comma replied:
overlay_image indeed has different arguments, mostly to let you use it without defining which argument you pass like in that example.
image goes: icon, LOC, icon_state, layer and dir. Usually you only need icon and icon_state for overlays, so that loc argument only gets in the way.

GinjaNinja32 replied:
iirc the loc argument on `image()` is optional

comma committed:
Things pointed out in PR

I don't thinl explicit layer setting is needed anyway, rest of overlays somehow layer fine without it.

comma committed:
Fixes accessories using onmob sprites when inhands or on inhand things

PsiOmegaDelta commented:
:+1:
